### PR TITLE
Offmap fixes and refactoring

### PR DESCRIPTION
**Pull request by sejtalk**
- Merged offmap decisions for sake of simplicity

### CODE REVIEW

**Comment by MemeWizard** [comment]
Request power triggers 2 times

<details>

image
image

</details>

**Comment by sejtalk** [comment]
> Request power triggers 2 times
> 
> image
> image
> 
> Looks like C'thun triggers no matter what OG your character serve
> 
> image

Fixed

**Comment by zumbak04** [comment]
You can request power already having power modifier.
<details>
<summary>Click to expand</summary>

image

</details>

File: decisions/wc_offmap_decisions.txt [hunk 1]
```diff
@@ -381,14 +427,18 @@ offmap_decisions = {
 			}
 		}
 	}
-	request_army_offmap_yoggsaron = {
+	request_general_offmap = {
```
Comment (zumbak04) [inline decisions/wc_offmap_decisions.txt]: I would comment it out since it's not used.

Comment (sejtalk) [reply decisions/wc_offmap_decisions.txt]: it will be changed in another branch

**Comment by sejtalk** [comment]
> You can request power already having power modifier.
> 
> Click to expand
> image

fixed

**Comment by zumbak04** [comment]
You can click on Request Power a few times and so collect two modifiers.
<details>
<summary>Click to expand</summary>

image
image

</details>

**Comment by MemeWizard** [comment]
My char don't have power, but decision is inactive.
Weird. I assume my char can't request power if any character already have power from that OG? Because My char can get powers from others OG.

<details>

image
image
image

</details>

**Comment by sejtalk** [comment]
> My char don't have power, but decision is inactive.
> Weird. I assume my char can't request power if any character already have power from that OG? Because my char can get powers from others OG, and nobody requested power from other OG yet.
> 
> image
> image
> image

Do you have requested_offmap_power character flag?

**Comment by MemeWizard** [comment]
>Do you have requested_offmap_power character flag?

Nope

**Comment by sejtalk** [comment]
> My char don't have power, but decision is inactive.
> Weird. I assume my char can't request power if any character already have power from that OG? Because my char can get powers from others OG, and nobody requested power from other OG yet.

fixed

File: decisions/wc_offmap_decisions.txt [hunk 2]
```diff
@@ -653,6 +661,10 @@ offmap_decisions = {
 					FROM = {
 						hidden_tooltip = { has_character_modifier = follower_of_yogg_saron}
 						has_offmap_currency = { offmap = offmap_yoggsaron value = 750 }
+						NOR = {
+							has_character_modifier = gaze_of_death
+							has_character_modifier = gaze_of_undeath
```
[the remark this answers is not in the document]
Comment (sejtalk) [reply decisions/wc_offmap_decisions.txt]: you can only request stuff from OG you worship now but not sure how to fix sacrifice decision

[the remark this answers is not in the document]
Comment (sejtalk) [reply decisions/wc_offmap_decisions.txt]: fixed in bug crusade branch

File: decisions/wc_offmap_decisions.txt [hunk 3]
```diff
@@ -636,108 +557,102 @@ offmap_decisions = {
...
-				hidden_effect = {
-					random_list = {
-						90 = { character_event = { id = WCOME.30 } }
-						10 = { character_event = { id = WCOME.31 } }
+			trigger_if = {
+				limit = { is_offmap_tag = offmap_cthun }
+				FROM = {
+					has_offmap_currency = { offmap = offmap_cthun value = 750 }
+					has_character_modifier = follower_of_cthun
+					NOR = {
```
Comment (zumbak04) [inline decisions/wc_offmap_decisions.txt]: Add scripted trigger.

File: decisions/wc_offmap_decisions.txt [hunk 4]
```diff
@@ -636,108 +557,102 @@ offmap_decisions = {
...
-		ai_will_do = {
-			factor = 1
-			modifier = {
-				factor = 0.3 # slow down
+			trigger_else_if = {
+				limit = { is_offmap_tag = offmap_nzoth }
+				FROM = {
+					has_offmap_currency = { offmap = offmap_nzoth value = 750 }
+					has_character_modifier = follower_of_nzoth
+					NOR = {
```
Comment (zumbak04) [inline decisions/wc_offmap_decisions.txt]: Add scripted trigger.

File: decisions/wc_offmap_decisions.txt [hunk 5]
```diff
@@ -636,108 +557,102 @@ offmap_decisions = {
...
-			is_landed = yes
-		}
-		allow = {
-			FROM = {
-				has_offmap_currency = { offmap = offmap_nzoth value = 750 }
+			trigger_else = {
+				FROM = {
+					has_offmap_currency = { offmap = offmap_yoggsaron value = 750 }
+					has_character_modifier = follower_of_yogg_saron
+					NOR = {
```
Comment (zumbak04) [inline decisions/wc_offmap_decisions.txt]: Add scripted trigger.

File: events/wc_offmap_events.txt [hunk 6]
```diff
@@ -633,7 +633,16 @@ character_event = {
 
 	option = {
 		name = EVTOPTA_WCOME.30
+		hidden_tooltip = {
```
Comment (zumbak04) [inline events/wc_offmap_events.txt]: Add scripted effect.

File: events/wc_offmap_events.txt [hunk 7]
```diff
@@ -649,7 +658,16 @@ character_event = {
 
 	option = {
 		name = EVTOPTA_WCOME.31
+		hidden_tooltip = {
```
Comment (zumbak04) [inline events/wc_offmap_events.txt]: Add scripted effect.

File: events/wc_offmap_events.txt [hunk 8]
```diff
@@ -665,7 +683,16 @@ character_event = {
 
 	option = {
 		name = EVTOPTA_WCOME.32
+		hidden_tooltip = {
```
Comment (zumbak04) [inline events/wc_offmap_events.txt]: Add scripted effect.

File: events/wc_offmap_events.txt [hunk 9]
```diff
@@ -681,7 +708,16 @@ character_event = {
 
 	option = {
 		name = EVTOPTA_WCOME.33
+		hidden_tooltip = {
```
Comment (zumbak04) [inline events/wc_offmap_events.txt]: Add scripted effect.

File: events/wc_offmap_events.txt [hunk 10]
```diff
@@ -697,7 +733,16 @@ character_event = {
 
 	option = {
 		name = EVTOPTA_WCOME.34
+		hidden_tooltip = {
```
Comment (zumbak04) [inline events/wc_offmap_events.txt]: Add scripted effect.

File: events/wc_offmap_events.txt [hunk 11]
```diff
@@ -713,7 +758,16 @@ character_event = {
 
 	option = {
 		name = EVTOPTA_WCOME.35
+		hidden_tooltip = {
```
Comment (zumbak04) [inline events/wc_offmap_events.txt]: Add scripted effect.

File: decisions/wc_offmap_decisions.txt (hunk 3 above)
Comment (sejtalk) [reply decisions/wc_offmap_decisions.txt]: fixed

File: decisions/wc_offmap_decisions.txt (hunk 4 above)
Comment (sejtalk) [reply decisions/wc_offmap_decisions.txt]: fixed

File: decisions/wc_offmap_decisions.txt (hunk 5 above)
Comment (sejtalk) [reply decisions/wc_offmap_decisions.txt]: fixed

File: events/wc_offmap_events.txt (hunk 6 above)
Comment (sejtalk) [reply events/wc_offmap_events.txt]: fixed

File: events/wc_offmap_events.txt (hunk 7 above)
Comment (sejtalk) [reply events/wc_offmap_events.txt]: fixed

File: events/wc_offmap_events.txt (hunk 8 above)
Comment (sejtalk) [reply events/wc_offmap_events.txt]: fixed

File: events/wc_offmap_events.txt (hunk 9 above)
Comment (sejtalk) [reply events/wc_offmap_events.txt]: fixed

File: events/wc_offmap_events.txt (hunk 10 above)
Comment (sejtalk) [reply events/wc_offmap_events.txt]: fixed

File: events/wc_offmap_events.txt (hunk 11 above)
Comment (sejtalk) [reply events/wc_offmap_events.txt]: fixed

File: events/wc_offmap_events.txt [hunk 12]
```diff
@@ -659,12 +654,7 @@ character_event = {
 	option = {
 		name = EVTOPTA_WCOME.31
 		hidden_tooltip = {
```
Comment (MemeWizard) [inline events/wc_offmap_events.txt]: It would be nice if this effect were not hidden.

Comment (sejtalk) [reply events/wc_offmap_events.txt]: you are not supposed to even have this modifer when using decision so it's better that way

**Comment by MemeWizard** [comment]
It would be nice if this condition has text

<details>

Безымянный2

</details>

You can make single "Don't already have any OG blessing" tooltip for these two conditions
<details>

Безымянный3

</details>

**Comment by sejtalk** [comment]
> It would be nice if this condition has text
> 
> You can make single "Don't already have any OG blessing" tooltip for these two conditions

I've hidden the second from scope

File: decisions/wc_offmap_decisions.txt [hunk 13]
```diff
@@ -580,7 +580,7 @@ offmap_decisions = {
 					lacks_offmap_power_trigger = yes
 				}
 			}
-			FROM = { NOT = { has_character_flag = requested_offmap_power } }
+			hidden_effect = { FROM = { NOT = { has_character_flag = requested_offmap_power } } }
```
Comment (MemeWizard) [inline decisions/wc_offmap_decisions.txt]: hidden_effect don't work in conditions

<details>

image

</details>

Comment (sejtalk) [reply decisions/wc_offmap_decisions.txt]: fixed

**Pull request merged**